tests/atomic.rs

```diff
@@ -0,0 +1,100 @@
+mod test_utilities;
+use async_rdma::{LocalMrReadAccess, Rdma};
+use std::{alloc::Layout, io, time::Duration};
+use test_utilities::test_server_client;
+
+mod cas {
+    use super::*;
+    mod cas_success {
+        use super::*;
+
+        async fn client(rdma: Rdma) -> io::Result<()> {
+            // alloc 8 bytes remote memory
+            let mut rmr = rdma.request_remote_mr(Layout::new::<[u8; 8]>()).await?;
+            let new_value = u64::from_le_bytes([1_u8; 8]);
+            // read, compare with rmr and swap `old_value` with `new_value`
+            rdma.atomic_cas(0, new_value, &mut rmr).await?;
+            // send rmr's meta data to the remote end
+            rdma.send_remote_mr(rmr).await?;
+            Ok(())
+        }
+
+        async fn server(rdma: Rdma) -> io::Result<()> {
+            // receive mr's meta data from client
+            let lmr = rdma.receive_local_mr().await?;
+            // assert the content of lmr, which was write by cas
+            let data = *lmr.as_slice();
+            assert_eq!(data, [1_u8; 8]);
+            tokio::time::sleep(Duration::from_secs(1)).await;
+            Ok(())
+        }
+
+        #[test]
+        fn main() {
+            test_server_client(server, client);
+        }
+    }
+
+    mod cas_wrong_align {
```

Comment:
Add a comment here to tell that it doesn't make any difference with wrong mr alignment, otherwise we're expecting an exception.

Reply:
I'm not sure what you mean.

Do you mean I should add a comment that "this test is the same as other tests except for wrong mr alignment"?

We can also put wrong cases in one test, but I think put them in dfiierent tests is more arresting, and of course more redundancy. Which do you think is better?

Reply:
This test tells users that even the alignment is not 8, it doesn't bother the cas operation. Because you put "wrong" in the mod name but there's no any exception or error in the test.

Reply:
It does bother the cas operation. The wrong_xxx cases have a #[should_panic] at their `main` function, and they will panic when they run to `atomic_cas`